### PR TITLE
Add support for wasm_import_module

### PR DESCRIPTION
**Pull request by Jasper-Bekkers**
For some of our wasm code we wanted to communicate with existing C bindings (and piggyback existing bindgen code), we needed this attribute supported from bindgen to make that work.

Without the attribute the linked code would end up in the default `env` namespace but for our use-case we needed to have these bindgen'd APIs in a different namespace.

I hope I've exposed the setting in the right way but feel free to let me know otherwise.

### CODE REVIEW

File: src/codegen/mod.rs [hunk 1]
```diff
@@ -3611,8 +3611,20 @@ impl CodeGenerator for Function {
...
+                quote! {
+                    #[link(wasm_import_module = #name)]
+                }
+            });
+
         let ident = ctx.rust_ident(canonical_name);
-        let tokens = quote!( extern #abi {
+        let tokens = quote!(
+            #wasm_link_attribute
+            extern #abi {
```
Comment (emilio) [inline src/codegen/mod.rs]: Could you reindent this too?

**Comment by emilio** [comment]
Also, could you squash your commits?

**Comment by Jasper-Bekkers** [comment]
Just squashed and re-indented.

File: src/codegen/mod.rs [hunk 2]
```diff
@@ -3611,10 +3611,22 @@ impl CodeGenerator for Function {
...
         let ident = ctx.rust_ident(canonical_name);
-        let tokens = quote!( extern #abi {
-            #(#attributes)*
-            pub fn #ident ( #( #args ),* ) #ret;
+        let tokens = quote!(
+            #wasm_link_attribute
+            extern #abi {
+                #(#attributes)*
+                pub fn #ident ( #( #args ),* ) #ret;
         });
```
Comment (emilio) [inline src/codegen/mod.rs]: This still looks misindented (as in, the closing brace doesn't align with the opening one), but not worth blocking on that.

**Pull request merged**